### PR TITLE
Subworkflow to create a phased vcf of somatic and germline variants for pVACseq

### PR DESCRIPTION
**Pull request by susannasiebert**
This hasn't been run on any data yet.

### CODE REVIEW

File: definitions/tools/pvacseq_combine_variants.cwl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Combine germline and somatic vcf for pVACseq phasing"
+baseCommand: ["/usr/bin/java", "-Xmx8g", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "CombineVariants"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 8000
+      tmpdirMin: 25000
+    - class: DockerRequirement
+      dockerPull: mgibio/cle
```
Comment (tmooney) [inline definitions/tools/pvacseq_combine_variants.cwl]: These could possibly use this smaller image instead:
```
dockerPull: "mgibio/gatk-cwl:3.6.0"
```

File: definitions/tools/sort_vcf.cwl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+label: "Sort VCF"
+
+baseCommand: ["/usr/bin/java", "-Xmx16g", "-jar", "/opt/picard/picard.jar", "SortVcf"]
+arguments:
+    ["O=", { valueFrom: $(runtime.outdir)/sorted.vcf }]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 16000
+    - class: DockerRequirement
+      dockerPull: "mgibio/cle"
```
Comment (tmooney) [inline definitions/tools/sort_vcf.cwl]: Similarly we have a Picard image:
```
dockerPull: "mgibio/picard-cwl:2.18.1"
```

File: definitions/tools/pvacseq_combine_variants.cwl [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Combine germline and somatic vcf for pVACseq phasing"
+baseCommand: ["/usr/bin/java", "-Xmx8g", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "CombineVariants"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 8000
```
Comment (tmooney) [inline definitions/tools/pvacseq_combine_variants.cwl]: We should make sure that the resource request on these is greater than the JVM memory request!

File: definitions/tools/read_backed_phasing.cwl [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Read-backed phasing"
+baseCommand: ["/usr/bin/java", "-Xmx8g", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "ReadBackedPhasing"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 8000
```
Comment (tmooney) [inline definitions/tools/read_backed_phasing.cwl]: Same comment here as on the other memory request.

File: definitions/tools/sort_vcf.cwl [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Sort VCF"
+
+baseCommand: ["/usr/bin/java", "-Xmx16g", "-jar", "/opt/picard/picard.jar", "SortVcf"]
+arguments:
+    ["O=", { valueFrom: $(runtime.outdir)/sorted.vcf }]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 16000
```
Comment (tmooney) [inline definitions/tools/sort_vcf.cwl]: Same comment here as on the other memory request.

**Comment by susannasiebert** [comment]
Huiming ran this workflow and found a few bugs that I fixed. She was able to run it successfully. Merging.

**Pull request merged**